src/config/validateVaultFiles.ts

```diff
@@ -34,31 +32,35 @@ const config = z.object({
   configs: z.record(z.optional(z.object({ activeProjectId: z.string() }))),
 })
 
-const version = z.object({
-  version: z.number(),
-})
-
 function isValidObject(objectType: any, object: any) {
   return objectType.safeParse(object).success
 }
 
-export function validateVaultFiles() {
-  // checking if extension installed for the first time
-  const checkOnlyVersionKeyInConfig =
-    isValidObject(version, configVault.getObject()) &&
-    Object.keys(configVault.getObject()).length === 1
+const configInlogin = z.object({
+  version: z.number(),
+  configs: z.record(z.optional(z.object({ activeProjectId: z.string() }))).optional(),
+})
+
+export function validateVaultFilesInLogin() {
+  const isConfigValid = isValidObject(configInlogin, configVault.getObject())
 
-  if (checkOnlyVersionKeyInConfig) {
-    return
+  if (!isConfigValid) {
+    clearVault()
   }
+}
 
+export function validateVaultFilesInAuth() {
   const isConfigValid = isValidObject(config, configVault.getObject())
   const isCredentialsValid = isValidObject(credentials, credentialsVault.getObject())
 
   if (!(isConfigValid && isCredentialsValid)) {
-    credentialsVault.clear()
-    configVault.clear()
-
-    window.showErrorMessage(configMessage.invalidConfigFiles)
+    clearVault()
   }
 }
```

Comment:
I'm a bit confused. What is "inLogin" and "inAuth" in this case?

Reply:
when initializing the authentication, I would like to check both the structure of config and credentials, so that in the schemas all the fields -except env- are necessary 
_the use case here if user already logged in with CLI, the structure of the vault file needs to be checked_

while loging in, only checking the config structure is enough to proceed
before that I was using the same function and when user logout and login again, since credentials file is empty, it couldn't pass validation, it was removing both files and asking to relogin again

I decided to differentiate two process